Add qubes.VMExec call, for running a single command

pwmarcz opened the pull request:
https://github.com/QubesOS/qubes-issues/issues/4850

marmarek commented on qubesagent/vmexec.py:
```diff
@@ -0,0 +1,56 @@
+
+    part = part.encode('ascii')
+
+    part = part.replace(b'--', b'/')
```
This could use a comment, especially why it's safe.

pwmarcz replied:
Right, it's actually not safe :( Things like `-2Fbin-2Fls` get converted to `/bin/ls` then replaced to `-bin-ls`. I found a better way.

marmarek commented on qubesagent/test_vmexec.py:
```diff
@@ -0,0 +1,38 @@
+        self.assertEqual(decode('-0a-0D'),
+                         [b'\n\r'])
+        self.assertEqual(decode('ls+--la'),
+                         [b'ls', b'-la'])
```
I'd add also a test like `ls+---61`

marmarek commented on qubesagent/test_vmexec.py:
```diff
@@ -0,0 +1,38 @@
+        with self.assertRaises(DecodeError):
+            decode('illegal/slash')
+        with self.assertRaises(DecodeError):
+            decode('illegal-singledash')
```
And here a test for incomplete encoded byte (at the end, and/or in the middle)

marmarek commented:
Also, besides commit messages, please also add issue reference(s) to the pull request description. Github the nicely list related PRs in the issue, including their state.

pwmarcz commented:
Updated:
- changed the approach to less hacky, fixing the `/` case
- permitted only capital letters in hex
- stronger tests (slashes, three/four dashes, small letters, incomplete byte)

marmarek commented on qubesagent/vmexec.py:
```diff
@@ -0,0 +1,58 @@
+
+    part = part.encode('ascii')
+
+    escape_re = br'--|-([A-F0-9]{2})'
```
re.compile?

pwmarcz replied:
Added.
